platform/viewer/src/components/ViewportGrid/ViewportGrid.js

```diff
@@ -0,0 +1,186 @@
+        className={classNames('viewport-container', {
+          active: activeViewportIndex === viewportIndex,
+        })}
+        key={viewportIndex}
+      >
+        <ViewportComponent
+          // Different "Viewport Types"? Feels too rigid to lock all into these props
+          {...viewportProps}
+          // We shouldn't need this?
+          // Used in `ConnectedCornerstoneViewport`
+          viewportIndex={viewportIndex}
+          setViewportActive={() => {
+            if (activeViewportIndex !== viewportIndex) {
+              setViewportActive(viewportIndex);
+            }
+          }}
+        />
+        {childrenWithProps}
+      </ViewportPane>
+    );
+  });
+
+  const rowSize = 100 / numRows;
+  const colSize = 100 / numColumns;
+
+  // http://grid.malven.co/
+  return (
+    <div
+      style={{
+        display: 'grid',
+        gridTemplateRows: `repeat(${numRows}, ${rowSize}%)`,
+        gridTemplateColumns: `repeat(${numColumns}, ${colSize}%)`,
+        height: '100%',
+        width: '100%',
+      }}
+    >
+      {magic}
+    </div>
+  );
+};
+
+const noop = () => {};
+
+ViewportGrid.propTypes = {
+  activeViewportIndex: PropTypes.number.isRequired,
+  numRows: PropTypes.number.isRequired,
+  numColumns: PropTypes.number.isRequired,
+  viewportPanes: PropTypes.arrayOf(
+    PropTypes.shape({
+      studyInstanceUid: PropTypes.string,
+      displaySetInstanceUid: PropTypes.string,
+      plugin: PropTypes.string,
+    })
+  ).isRequired,
+  availablePlugins: PropTypes.object.isRequired,
+  setViewportData: PropTypes.func,
+  setViewportActive: PropTypes.func, // Connected
+  studies: PropTypes.array,
+  children: PropTypes.node,
+};
+
+ViewportGrid.defaultProps = {
+  activeViewportIndex: 0,
+  availablePlugins: {
+    DefaultViewport,
+  },
+  setViewportActive: noop,
+  defaultPlugin: 'DefaultViewport',
+};
+
+/**
+ *
+ *
+ * @param {*} studies
+ * @param {string} studyInstanceUid
+ * @param {string} displaySetInstanceUid
+ * @returns
+ */
+function _getDisplaySetStackFromStudies(
+  studies,
+  studyInstanceUid,
+  displaySetInstanceUid
+) {
+  if (!studyInstanceUid || !displaySetInstanceUid) {
+    return undefined;
+  }
+
+  const study = studies.find(
+    study => study.studyInstanceUid === studyInstanceUid
+  );
+  const studyDisplaySet = study.displaySets.find(set => {
+    return set.displaySetInstanceUid === displaySetInstanceUid;
+  });
+
+  // Get stack from Stack Manager
+  // TODO: Study contains this display set. Why do we need to pass both? Should just take displaySetInstanceUid?
+  // TODO: The ergonomics of this are weird -- What is createStacks doing to study.displaySets[displaySetInstanceUid]?
+  const storedStack = StackManager.findOrCreateStack(study, studyDisplaySet);
+
+  if (storedStack) {
+    return {
+      // studyInstanceUid,
+      // displaySetInstanceUid,
+      imageIds: storedStack.imageIds,
+      frameRate: storedStack.frameRate,
+    };
+  }
+
+  return undefined;
```

Comment:
Useless return?

Reply:
I could keep it or leave it. It's reminder that we're returning a value (undefined) if `storedStack` is undefined